index.js

```diff
@@ -274,78 +274,75 @@
   }
 
   //  :: Type
-  var a = $.TypeVariable('a');
-  var b = $.TypeVariable('b');
-  var c = $.TypeVariable('c');
-  var d = $.TypeVariable('d');
-  var e = $.TypeVariable('e');
-  var g = $.TypeVariable('g');
-  var l = $.TypeVariable('l');
-  var r = $.TypeVariable('r');
+  var a = $.TypeVariable ('a');
+  var b = $.TypeVariable ('b');
+  var c = $.TypeVariable ('c');
+  var d = $.TypeVariable ('d');
+  var e = $.TypeVariable ('e');
+  var g = $.TypeVariable ('g');
+  var l = $.TypeVariable ('l');
+  var r = $.TypeVariable ('r');
 
   //  :: Type -> Type
-  var f = $.UnaryTypeVariable('f');
-  var m = $.UnaryTypeVariable('m');
-  var t = $.UnaryTypeVariable('t');
-  var w = $.UnaryTypeVariable('w');
+  var f = $.UnaryTypeVariable ('f');
+  var m = $.UnaryTypeVariable ('m');
+  var t = $.UnaryTypeVariable ('t');
+  var w = $.UnaryTypeVariable ('w');
 
   //  :: Type -> Type -> Type
-  var p = $.BinaryTypeVariable('p');
-  var s = $.BinaryTypeVariable('s');
+  var p = $.BinaryTypeVariable ('p');
+  var s = $.BinaryTypeVariable ('s');
 
   //  eitherTypeIdent :: String
   var eitherTypeIdent = 'sanctuary/Either';
 
   //  $Either :: Type -> Type -> Type
-  var $Either = $.BinaryType(
-    eitherTypeIdent,
-    readmeUrl('EitherType'),
-    typeEq(eitherTypeIdent),
-    function(either) { return either.isLeft ? [either.value] : []; },
-    function(either) { return either.isRight ? [either.value] : []; }
-  );
+  var $Either = $.BinaryType
+    (eitherTypeIdent)
+    (readmeUrl ('EitherType'))
+    (typeEq (eitherTypeIdent))
+    (either (of (Array)) (K ([])))
+    (either (K ([])) (of (Array)));
```

Comment:
Now that functions such as `K` are curried (and hoisted) we're able to use them throughout the file. :)